src/main/java/com/google/cloud/spanner/r2dbc/SpannerStatement.java

```
@@ -93,6 +94,7 @@ public Statement bindNull(int i, Class<?> type) {
     // only a single constructor that takes a flux of rows and a mono of the # rows updated).
     return Mono
         .just(new SpannerResult(
-            Flux.create(sink -> result.subscribe(new PartialResultFluxConverter(sink)))));
+            Flux.create(sink -> result.subscribe(new PartialResultFluxConverter(sink))),
```

Does result need to be cached here? i.e. if user calls `map` twice, I think the query will get run twice for the same reason that `beginTransaction` was run twice.

---

I don't think we could cache the flux, since that may be huge. Last time we checked with Mark he said we could just throw an error since result sets aren't meant to be mapped through once than once.

---

Cool, got it.